reconcile/utils/terraform_resource_spec.py

```diff
@@ -0,0 +1,123 @@
+from dataclasses import dataclass, field
+import json
+from typing import Any, Optional, cast
+import base64
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+
+
+@dataclass
+class TerraformResourceSpec:
+
+    resource: dict[str, Any]
+    namespace: dict[str, Any]
+    secret: dict[str, str] = field(init=False, default_factory=lambda: {})
+
+    @property
+    def provider(self):
+        return self.resource.get("provider")
+
+    @property
+    def identifier(self):
+        return self.resource.get("identifier")
+
+    @property
+    def account(self):
+        return self.resource.get("account")
+
+    @property
+    def namespace_name(self) -> str:
+        return self.namespace["name"]
+
+    @property
+    def cluster_name(self) -> str:
+        return self.namespace["cluster"]["name"]
+
+    @property
+    def output_prefix(self):
+        return f"{self.identifier}-{self.provider}"
+
+    @property
+    def output_resource_name(self):
+        return self.resource.get("output_resource_name") or self.output_prefix
+
+    @property
+    def annotations(self) -> dict[str, str]:
+        annotation_str = self.resource.get("annotations")
+        if annotation_str:
+            return json.loads(annotation_str)
+        else:
+            return {}
+
+    def get_secret_field(self, field: str) -> Optional[str]:
+        return self.secret.get(field)
+
+    def id_object(self) -> "TerraformResourceIdentifier":
+        return TerraformResourceIdentifier.from_dict(self.resource)
+
+    def construct_oc_resource(self, integration: str, integration_version: str) -> OR:
```

Comment:
if this is building a secret, why not something like?
```suggestion
    def construct_oc_secret(self, integration: str, integration_version: str) -> OR:
```
or even using the more common `build`
```suggestion
    def build_oc_secret(self, integration: str, integration_version: str) -> OR:
```

Reply:
i used the second name suggestion - thx! [b023c96](https://github.com/app-sre/qontract-reconcile/pull/2331/commits/b023c965c1dced92eb46ce03f7e2cb1343cfe509)